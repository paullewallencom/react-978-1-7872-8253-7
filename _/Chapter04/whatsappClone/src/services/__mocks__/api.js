export const getMockDataSpy = jest.fn();

export const getMockData = () => { 
    return new Promise(resolve => getMockDataSpy(() => resolve([{}]))) 
}