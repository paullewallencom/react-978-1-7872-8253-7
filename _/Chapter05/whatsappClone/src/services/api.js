export const mockMessages = [
    {
        incoming: true,
        message: 'Hi Vladimir'
    },
    {
        incoming: false,
        message: 'Hi, John'
    },
    {
        incoming: true,
        message: 'When we will learn real data fetching'
    },
    {
        incoming: false,
        message: `First we need to learn debugging and testing, then we will learn styling and animations and then we will learn real data fetching. Let's use this mock data for now ok?`
    }
]

export const getMockData = () => (
    new Promise(resolve => setTimeout(() => resolve(mockMessages), 1000))
)