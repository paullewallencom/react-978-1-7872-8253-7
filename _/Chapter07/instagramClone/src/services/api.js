import { signup, login, initialize } from './firebase';


export const initApi = () => initialize();


export {
  login,
  signup
}

export const getMessages = (updaterFn) => setListener('messages', updaterFn);


export const postMessage = (message) => {
  if (Boolean(message)) {
    pushData('messages', {
      incoming: false,
      message
    })
  }
}
