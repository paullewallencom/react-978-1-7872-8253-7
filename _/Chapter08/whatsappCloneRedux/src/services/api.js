import { setListener, pushData, initialize } from './firebase';

const mockMessages = [
  {
    incoming: true,
    message: 'Hi Vladimir'
  },
  {
    incoming: false,
    message: 'Hi John'
  },
  {
    incoming: true,
    message: 'When we will learn real data fetching'
  },
  {
    incoming: false,
    message: `First we need to learn debugging and testing, then we will learn styling and animations and then we will learn real data fetching. Let's use mock data for now ok?`
  }
]

export const getMockData = () => (
  new Promise(resolve => setTimeout(() => resolve(mockMessages), 1000))
)

export const initApi = () => initialize();


export const getMessages = (updaterFn) => setListener('messages', updaterFn);


export const postMessage = (message) => {
  if (Boolean(message)) {
    return pushData('messages', {
      incoming: false,
      message
    })
  }
}
