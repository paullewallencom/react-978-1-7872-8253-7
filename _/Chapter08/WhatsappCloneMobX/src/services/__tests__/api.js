import 'react-native';
import React from 'react';
import { getMockData } from '../api';


it('resolves to mock data',async () => {
  const messages = await getMockData();
  expect(messages).toMatchSnapshot();
});
