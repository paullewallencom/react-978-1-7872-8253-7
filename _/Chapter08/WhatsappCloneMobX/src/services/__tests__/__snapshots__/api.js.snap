// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolves to mock data 1`] = `
Array [
  Object {
    "incoming": true,
    "message": "Hi Vladimir",
  },
  Object {
    "incoming": false,
    "message": "Hi John",
  },
  Object {
    "incoming": true,
    "message": "When we will learn real data fetching",
  },
  Object {
    "incoming": false,
    "message": "First we need to learn debugging and testing, then we will learn styling and animations and then we will learn real data fetching. Let's use mock data for now ok?",
  },
]
`;
